### PR TITLE
bpo-41700: Skip test if the locale is not supported

### PR DESCRIPTION
**Pull request by corona10**
https://bugs.python.org/issue41700

### CODE REVIEW

File: Lib/test/test_c_locale_coercion.py [hunk 1]
```diff
@@ -405,7 +405,11 @@ def test_LC_ALL_set_to_C(self):
 
     def test_PYTHONCOERCECLOCALE_set_to_one(self):
         # skip the test if the LC_CTYPE locale is C or coerced
-        old_loc = locale.setlocale(locale.LC_CTYPE, None)
+        try:
+            old_loc = locale.setlocale(locale.LC_CTYPE, None)
```
Comment (vstinner) [inline Lib/test/test_c_locale_coercion.py]: This call works, it just gets the current locale.

locale.setlocale(locale.LC_CTYPE, "") is the one which fails: it sets LC_CTYPE to the user preferred locale.

**Review by vstinner — COMMENTED** [review]
Your PR doesn't fix the issue, see https://bugs.python.org/issue41700#msg376320 to reproduce the issue.

**Review by corona10 — COMMENTED** [review]
@vstinner 
Thanks, Victor
I fixed the wrong point
I 've updated the PR and it works well on the reproducing scenario

**Review by vstinner — APPROVED** [review]
LGTM. I tested manually that the PR fix https://bugs.python.org/issue41700.

It should be backport to 3.9, and maybe also to 3.8, no?

**Comment by corona10** [comment]
> It should be backport to 3.9, and maybe also to 3.8, no?

I will add the label! Thanks!

**Pull request merged**